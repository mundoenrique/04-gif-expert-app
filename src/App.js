import { useState } from 'react';

import AddCategory from './components/AddCategory';
import GifGrid from './components/GifGrid';

function App({ defaultCat = ['One Punch'] }) {
	const [categories, setCategories] = useState(defaultCat);

	return (
		<>
			<h2>GifExpertApp</h2>
			<AddCategory setCategories={setCategories} />
			<hr />
			{
				categories.map(category => (
					<GifGrid
						key={category}
						category={category}
					/>
				))
			}

		</>
	);
}

export default App;
