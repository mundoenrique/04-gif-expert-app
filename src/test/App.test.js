import { shallow } from "enzyme";
import App from "../App";

describe('Pruebas sobre el cmponente principal <App />', () => {

	test('Debe mostrar el componente <App /> correctamente', () => {
		const wrapper = shallow(<App />);

		expect(wrapper).toMatchSnapshot();
	});
	test('Debe mostrar la lista de categorias', () => {
		const categories = ['One Punch', 'Dragon Ball'];
		const wrapper = shallow(<App defaultCat={categories} />);

		expect(wrapper.find('GifGrid').length).toBe(categories.length);
	});
});
