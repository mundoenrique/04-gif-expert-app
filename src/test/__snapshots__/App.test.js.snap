// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas sobre el cmponente principal <App /> Debe mostrar el componente <App /> correctamente 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <GifGrid
    category="One Punch"
    key="One Punch"
  />
</Fragment>
`;
